mav_trajectory_generation/src/trajectory.cpp

```diff
@@ -271,6 +283,43 @@ Vertex Trajectory::getGoalVertex(int max_derivative_order) const {
   return getVertexAtTime(max_time_, max_derivative_order);
 }
 
+bool Trajectory::getVertices(int max_derivative_order_pos,
+                             int max_derivative_order_yaw,
+                             Vertex::Vector* pos_vertices,
+                             Vertex::Vector* yaw_vertices) const {
+  CHECK_NOTNULL(pos_vertices);
+  CHECK_NOTNULL(yaw_vertices);
+  const std::vector<size_t> kPosDimensions = {0, 1, 2};
+  const std::vector<size_t> kYawDimensions = {3};
+  const int kMaxDerivativeOrder =
+      std::max(max_derivative_order_pos, max_derivative_order_yaw);
+  pos_vertices->resize(segments_.size() + 1, Vertex(3));
+  yaw_vertices->resize(segments_.size() + 1, Vertex(1));
+
+  // Start vertex.
+  Vertex temp_vertex(4);
```

Comment:
Will this work for D==6?

Reply:
nope! added new function overload for generic case: [20bdcd8](https://github.com/ethz-asl/mav_trajectory_generation/pull/97/commits/93cdeebbf39664a71e5368c128eb478bc21f23b9)